go/vt/vttablet/tabletmanager/vreplication/vreplicator.go

```diff
@@ -406,6 +429,38 @@ func (vr *vreplicator) resetFKCheckAfterCopy() error {
 	return err
 }
 
+func (vr *vreplicator) setSQLMode() (func(), error) {
+	// First save the original SQL mode if we have not already done so
+	if vr.originalSQLMode == "" {
+		res, err := vr.dbClient.Execute(getSQLModeQuery)
+		if err != nil || len(res.Rows) != 1 {
+			return nil, fmt.Errorf("could not get the original sql_mode on target: %v", err)
+		}
+		vr.originalSQLMode = res.Rows[0][0].ToString()
```

Comment:
Opinion: we really should move away from `[0][0]`. It's archaic. Instead, you may:
```suggestion
		vr.originalSQLMode = res.Named().Row().AsString("sql_mode", "")
```